test/recipes/70-test_tls13messages.t

```diff
@@ -296,4 +301,61 @@ checkhandshake($proxy, checkhandshake::HRR_RESUME_HANDSHAKE,
                | checkhandshake::PSK_CLI_EXTENSION
                | checkhandshake::PSK_SRV_EXTENSION,
                "Resumption handshake with HRR test");
+
+#Test 16: signature_algorithms with 1.3-only ClientHello
+my $testtype = PURE_SIGALGS;
+my $dsa_status = 1;
+$proxy->clear();
+$proxy->clientflags("-tls1_3");
+$proxy->filter(\&modify_sigalgs_filter);
+$proxy->start();
+ok($dsa_status, "DSA/SHA2 sigalg sent for 1.3-only ClientHello");
+
+#Test 17: signature_algorithms with backwards compatible ClientHello
+$testtype = COMPAT_SIGALGS;
+$dsa_status = 1;
```

Comment:
Perhaps.  It's probably best to avoid a pattern where we set it to 0 before the test, and the filter sets it to 1 at the start of the filter but can reset it back to 0 on error, though.  Which means the first test would need some extra local variable to keep track of state the way the second one does right now (but could be avoided).
I guess, on the whole, it's still worth doing.